### PR TITLE
osdc/Objecter.cc: Initialize objects with nullptr

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/osdc/Objecter.cc [hunk 1]
```diff
@@ -1319,18 +1320,20 @@ void Objecter::handle_osd_map(MOSDMap *m)
...
        p != need_resend_linger.end(); ++p) {
     LingerOp *op = *p;
     if (!op->session) {
       _calc_target(&op->target, nullptr);
-      OSDSession *s = NULL;
+      OSDSession *s = nullptr;
       const int r = _get_session(op->target.osd, &s, sul);
       assert(r == 0);
-      assert(s != NULL);
+      assert(s != nullptr);
```
Comment (tchaikov) [inline src/osdc/Objecter.cc]: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions

File: src/osdc/Objecter.cc [hunk 2]
```diff
@@ -3401,7 +3412,7 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
     _session_op_remove(s, op);
     sl.unlock();
     put_session(s);
-
+    s = nullptr;
```
Comment (amitkumar50) [inline src/osdc/Objecter.cc]: NIT:tab missed

**Comment by tchaikov** [comment]
needs rebase.

**Comment by liewegas** [comment]
rebase, please!

**Comment by jdurgin** [comment]
closing since it's a cleanup but would need rebasing.

**Pull request closed without merging**